### PR TITLE
Upgrade versions and add extra content

### PR DESCRIPTION
**Pull request by radix0000**
Has been upgraded as far as is possible (while sticking with sphinx-rtd-theme). Versions of important packages are:

sphinx==5.3.0
sphinx-rtd-theme==1.1.1
myst-parser==0.18.1
docutils==0.17.1

Kitchen sink and sphinx examples for tables etc. added.

### CODE REVIEW

File: docs/examples/admonitions.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    }
+    .admonition.pied-piper > .admonition-title {
+      background-color: rgba(43, 155, 70, 0.1);
+      border-color: rgb(43, 155, 70);
+    }
+    .admonition.pied-piper > .admonition-title::before {
+      background-color: rgb(43, 155, 70);
+      -webkit-mask-image: var(--icon--pied-piper);
+      mask-image: var(--icon--pied-piper);
+    }
```
Comment (Bjwebb) [inline docs/examples/admonitions.rst]: This css doesn't seem to be used anywhere, so the admonition at the bottom doesn't get rendered correctly https://sphinx-base.readthedocs.io/en/upgrade_to_latest_version/examples/admonitions.html

**Review by Bjwebb — CHANGES_REQUESTED** [review]

**Comment by Bjwebb** [comment]
I've spotted something that seems to be broken with the examples.

I'm happy with all the version updates though, so do make those same changes to [openreferral/specification](https://github.com/openreferral/specification/).

**Comment by radix0000** [comment]
Have added back missing CSS file, and custom admonition at bottom should now be rendering correctly.

**Review by Bjwebb — APPROVED** [review]
Approving now because admonition is now clearly styled. 

The logo looks to be slightly mangled on my screen though:
Screenshot from 2023-01-19 17-43-46

**Pull request merged**